### PR TITLE
feat: compute pool name is editable

### PR DESCRIPTION
**Pull request by Guttz**
Release Notes
-------------

Checklist
---------
1. [CRUCIAL] Is the change for features that are already live in prod?
   * yes: ok

What
----

Name of a flink compute pool is now editable

References
----------

Test & Review
-------------

Open Questions / Follow-ups
---------------------------

### CODE REVIEW

File: internal/cmd/flink/command_compute_pool_update.go [hunk 1]
```diff
@@ -20,18 +20,17 @@ func (c *command) newComputePoolUpdateCommand() *cobra.Command {
 		RunE:              c.computePoolUpdate,
 		Example: examples.BuildExampleString(
 			examples.Example{
-				Text: `Update Flink compute pool "my-compute-pool" to 2 CFUs.`,
-				Code: "confluent flink compute-pool update my-compute-pool --cfu 2",
+				Text: `Update name and CFU count of a Flink compute pool.`,
+				Code: "confluent flink compute-pool update my-compute-pool --name new_name --cfu 2",
```
Comment (brianstrauch) [inline internal/cmd/flink/command_compute_pool_update.go:24]:
```suggestion
				Code: `confluent flink compute-pool update my-compute-pool --name "new name" --cfu 2`,
```

File: internal/cmd/flink/command_compute_pool_update.go [hunk 2]
```diff
@@ -79,6 +78,14 @@ func (c *command) computePoolUpdate(cmd *cobra.Command, args []string) error {
 		update.Spec.MaxCfu = flinkv2.PtrInt32(cfu)
 	}
 
+	name, err := cmd.Flags().GetString("name")
```
Comment (brianstrauch) [inline internal/cmd/flink/command_compute_pool_update.go:81]: Do you mind adding the following to the beginning of the function? We want to make sure at least one of `--name` or `--cfu` is passed.
```
if err := errors.CheckNoUpdate(cmd.Flags(), "name", "cfu"); err != nil {
	return err
}
```

File: internal/cmd/flink/command_compute_pool_update.go [hunk 3]
```diff
@@ -20,18 +20,17 @@ func (c *command) newComputePoolUpdateCommand() *cobra.Command {
...
-				Text: `Update Flink compute pool "my-compute-pool" to 2 CFUs.`,
-				Code: "confluent flink compute-pool update my-compute-pool --cfu 2",
+				Text: `Update name and CFU count of a Flink compute pool.`,
+				Code: "confluent flink compute-pool update my-compute-pool --name new_name --cfu 2",
 			},
 		),
 	}
 
 	cmd.Flags().Int32("cfu", 0, "Number of Confluent Flink Units (CFU).")
+	cmd.Flags().String("name", "", "Name of the compute pool.")
```
Comment (brianstrauch) [inline internal/cmd/flink/command_compute_pool_update.go:30]: Nit: `--name` normally comes first.
```suggestion
	cmd.Flags().String("name", "", "Name of the compute pool.")
	cmd.Flags().Int32("cfu", 0, "Number of Confluent Flink Units (CFU).")
```

Comment (brianstrauch) [reply internal/cmd/flink/command_compute_pool_update.go:30]: I wonder if it would help to use `uint32` for `--cfu` here.

**Review by brianstrauch — APPROVED** [review]
thanks!!

**Review by luca-filipponi — APPROVED** [review]

**Pull request merged**